src/actions/sources.js

```diff
@@ -331,14 +335,13 @@ export function togglePrettyPrint(sourceId: string) {
     }
 
     if (prettySource) {
-      return dispatch(selectSource(prettySource.get("id"), options));
+      return dispatch(selectLocation(options.location));
     }
 
     const newPrettySource = await dispatch(createPrettySource(sourceId));
     await dispatch(remapBreakpoints(sourceId));
     await dispatch(setEmptyLines(newPrettySource.id));
-
-    return dispatch(selectSource(newPrettySource.id, options));
+    return dispatch(selectLocation({ sourceId: newPrettySource.id }));
```

Comment:
#4573 ?

Reply:
looks like this might solve that one too!